src/saltext/vmware/states/vmc_dhcp_profiles.py

```diff
@@ -0,0 +1,390 @@
+        is mandatory when type is relay.
+
+    tags
+        (Optional) Opaque identifiers meaningful to the user.
+
+        .. code::
+
+            tags:
+              - tag: <tag-key-1>
+                scope: <tag-value-1>
+              - tag: <tag-key-2>
+                scope: <tag-value-2>
+
+    lease_time
+        (Optional) IP address lease time in seconds. Minimum value is 60.
+        Maximum value is 4294967295. Default value is 86400
+        Note: This field is applicable only when type is server
+
+    display_name
+        Identifier to use when displaying entity in logs or GUI.
+
+    Example values:
+
+        .. code::
+
+            for dhcp-server-profiles:
+
+            server_addresses:
+              - 10.22.12.2/23
+            tags:
+              - tag: tag1
+                scope: scope1
+            lease_time: 86400
+
+            for dhcp-relay-profiles:
+
+            server_addresses:
+              - 10.1.1.1
+            tags:
+              - tag: tag1
+                scope: scope1
+
+    """
+
+    input_dict = {
+        "server_addresses": server_addresses,
+        "tags": tags,
+        "lease_time": lease_time,
+        "display_name": display_name,
+    }
+
+    if type == vmc_constants.RELAY and lease_time:
+        return vmc_state._create_state_response(
+            name=name, comment="lease_time is not applicable for DHCP Relay Profile", result=False
+        )
+
+    if type == vmc_constants.RELAY:
+        input_dict.pop("lease_time")
+
+    input_dict = {k: v for k, v in input_dict.items() if v != vmc_constants.VMC_NONE}
+
+    dhcp_profile = __salt__["vmc_dhcp_profiles.get_by_id"](
+        hostname=hostname,
+        refresh_key=refresh_key,
+        authorization_host=authorization_host,
+        org_id=org_id,
+        sddc_id=sddc_id,
+        type=type,
+        dhcp_profile_id=dhcp_profile_id,
+        verify_ssl=verify_ssl,
+        cert=cert,
+    )
+
+    if "error" in dhcp_profile:
+        if "could not be found" in dhcp_profile["error"]:
+            dhcp_profile = None
+        else:
+            return vmc_state._create_state_response(
+                name=name, comment=dhcp_profile["error"], result=False
+            )
+
+    if __opts__.get("test"):
+        log.info("present is called with test option")
+        if dhcp_profile:
+            return vmc_state._create_state_response(
+                name=name,
+                comment="State present will update DHCP Profile {}".format(dhcp_profile_id),
+            )
+        else:
+            return vmc_state._create_state_response(
+                name=name,
+                comment="State present will create DHCP Profile {}".format(dhcp_profile_id),
+            )
+
+    if dhcp_profile:
+        updatable_keys = input_dict.keys()
+        is_update_required = vmc_state._check_for_updates(
+            dhcp_profile, input_dict, updatable_keys, ["tags"]
+        )
```

Comment:
**question** I don't know if I asked this question before, but if we're passing in ``input_dict`` and ``input_dict`` as keys, why not in _check_for_updates just do this?

```
updateable_keys = updateable_keys or input_dict.keys()
```

Reply:
updated